### ISSUE TITLE
[Bug]: [restful v2]value in elementTypeParams should be integer, but it can only be str in current implement

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#31176]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

for example
```
                    {"fieldName": "varchar_array", "dataType": "Array", "elementDataType": "VarChar",
                     "elementTypeParams": {"max_capacity": "1024", "max_length": "256"}},
```

### Expected Behavior

```
                    {"fieldName": "varchar_array", "dataType": "Array", "elementDataType": "VarChar",
                     "elementTypeParams": {"max_capacity": 1024, "max_length": 256}},
```

### Anything else?

this issue also happened in indexConfig
`payload["indexParams"][0]["indexConfig"]["nlist"] = "16384"`

**Comment by PowderLi** [comment]
keep the same with grpc definition: [schema.proto](https://github.com/milvus-io/milvus-proto/blob/master/proto/schema.proto#L58)

```
message FieldSchema {
  ...
  repeated common.KeyValuePair type_params = 6;
  ...
}
```

**Comment by zhuwenxing** [comment]
The focus of this issue is more on user-friendliness because these parameters, intuitively speaking, should actually be of integer type.

**Comment by PowderLi** [comment]
if a new type param was added, whose value may not be integer, in the future?

**Comment by zhuwenxing** [comment]
/assign @PowderLi 
I think it would be better to have a resolution for this issue

**Comment by zhuwenxing** [comment]
@PowderLi 
![image](https://github.com/milvus-io/milvus/assets/12268675/347e191a-b754-400e-b2cb-688ec74af6ce)

the value in fields can be int now, but the value is params still needs to be str

**Comment by zhuwenxing** [comment]
`shardsNum`, `partitionsNum`, and `ttlSeconds` expect to be int, but require to be str

```
method: post, 
url: http://10.104.24.9:19530/v2/vectordb/collections/create, 
cost time: 0.08028078079223633, 
header: {'Content-Type': 'application/json', 'Authorization': 'Bearer root:Milvus', 'RequestId': 'b18401dc-fe39-11ee-ae6d-acde48001122', 'Request-Timeout': '1'}, 
payload: {"collectionName": "test_collection_2024_04_19_18_43_42_984513YMatqSHZ", "enableDynamicField": true, "params": {"shardsNum": "2", "partitionsNum": 36, "consistencyLevel": "Strong", "ttlSeconds": "360"}, "schema": {"fields": [{"fieldName": "book_id", "dataType": "Int64", "isPrimary": true, "elementTypeParams": {}}, {"fieldName": "word_count", "dataType": "Int64", "isPartitionKey": true, "elementTypeParams": {}}, {"fieldName": "book_describe", "dataType": "VarChar", "elementTypeParams": {"max_length": 256}}, {"fieldName": "json", "dataType": "JSON", "elementTypeParams": {}}, {"fieldName": "int_array", "dataType": "Array", "elementDataType": "Int64", "elementTypeParams": {"max_capacity": "1024"}}, {"fieldName": "book_intro", "dataType": "FloatVector", "elementTypeParams": {"dim": 128}}]}, "indexParams": [{"fieldName": "book_intro", "indexName": "book_intro_vector", "metricType": "COSINE"}]}, 
response: {"code":1801,"message":"can only accept json format request, error: json: cannot unmarshal number into Go struct field CollectionReq.params of type string"} (milvus.py:20)
```

`max_length` expects to be int, but requires to be str
```
method: post, 
url: http://10.104.24.9:19530/v2/vectordb/collections/create, 
cost time: 0.0254971981048584, 
header: {'Content-Type': 'application/json', 'Authorization': 'Bearer root:Milvus', 'RequestId': '6d9e7686-fe3a-11ee-b378-acde48001122', 'Request-Timeout': '1'}, 
payload: {"collectionName": "test_collection_2024_04_19_18_48_58_567290qthJXGxH", "dimension": 128, "metricType": "L2", "primaryFieldName": "id", "vectorFieldName": "vector", "idType": "VarChar", "params": {"max_length": 256, "consistencyLevel": "Strong"}}, 
response: {"code":1801,"message":"can only accept json format request, error: json: cannot unmarshal number into Go struct field CollectionReq.params of type string"} (milvus.py:20)
```

`Request-Timeout` in headers expects to be int, but requires to be str
```
testcases/test_collection_operations.py:19 (TestCreateCollection.test_create_collections_quick_setup[128])
header = ('Request-Timeout', 1)

    def check_header_validity(header):
        """Verifies that header value is a string which doesn't contain
        leading whitespace or return characters. This prevents unintended
        header injection.
    
        :param header: tuple, in the format (name, value).
        """
        name, value = header
    
        if isinstance(value, bytes):
            pat = _CLEAN_HEADER_REGEX_BYTE
        else:
            pat = _CLEAN_HEADER_REGEX_STR
        try:
>           if not pat.match(value):
E           TypeError: expected string or bytes-like object, got 'int'
```

**Comment by zhuwenxing** [comment]
verified and fixed with master-20240424-dcc15e3e-amd64
